packages/daemon/src/wallet/LedgerClient.js

```diff
@@ -0,0 +1,65 @@
+// @flow
+
+import Transport from '@ledgerhq/hw-transport-node-hid'
+import LedgerAppEth from '@ledgerhq/hw-app-eth'
+
+const ITEMS_PER_PAGE = 10
+const HD_PATH = "m/44'/60'/0'/"
+let ledgerApp
+
+const getLedgerApp = async (): Promise<LedgerAppEth> => {
+  if (ledgerApp) {
+    return ledgerApp
+  }
+  const transport = await Transport.create()
+  ledgerApp = new LedgerAppEth(transport)
+  return ledgerApp
+}
+
+export const getAddressAtIndex = async (params: {
+  index: number,
+}): Promise<string> => {
+  const app = await getLedgerApp()
+  const path = HD_PATH + params.index
+  const account = await app.getAddress(path)
+  return account.address
+}
+
+export const getAccounts = async (params: {
+  from: number,
+  to: number,
+}): Promise<Array<string>> => {
+  const app = await getLedgerApp()
+  const accounts = []
+  for (let i = params.from; i < params.to; i++) {
+    const path = HD_PATH + i
+    const account = await app.getAddress(path)
```

Comment:
It's not ideal to have an `await` in a loop as it makes all the calls happen in series rather than parallel.
Here the logic could be to first extract the paths in the loop, and then use `const accounts = await Promise.all(paths.map(app.getAddress))`.